src/Commands/CreateOrEditTaskParser.ts

```diff
@@ -89,9 +117,9 @@ export const taskFromLine = ({ line, path }: { line: string; path: string }): Ta
         blockLink,
         priority: Priority.None,
         createdDate,
-        startDate: null,
-        scheduledDate: null,
-        dueDate: null,
+        startDate: startDateOrNull,
+        scheduledDate: scheduledDateOrNull,
+        dueDate: dueDateOrNull,
```

Comment:
There is indeed. After I wrote it I noticed that I'm mimicking the `Task.fromLine()` implementation with the sole difference that I'm not checking the Global Filter presence.

So I'm inclined to add an optional parameter to skip the GlobalFilter check.

Or just remove that check as well as this whole "task reconstruction" part. It feels like it will be easier. But then other issues pop up. So the first solution seems to be simpler. But the optionals are so smelly...

Dunno, I will ponder over this. I'm buying any thought you might have as well =)

Reply:
And I'm concious that I should've started by writing a test but the refactorings here were pretty obvious and most importantly I feel like I'm going to rework the tests anyways.